### PR TITLE
feat(select): options search

### PR DESCRIPTION
**Pull request by sashaqred**
### Please read and mark the following check list before creating a pull request:

- [x] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/nebular/blob/master/CONTRIBUTING.md) guide.

### CODE REVIEW

File: src/framework/theme/components/select/select.component.html [hunk 1]
```diff
@@ -1,4 +1,5 @@
 <button
+  [hidden]="isOptionSearchInputAllowed"
```
Comment (Tibing) [inline src/framework/theme/components/select/select.component.html:2]: why is it hidden? why not *ngIf?

File: src/framework/theme/components/select/select.component.ts [hunk 2]
```diff
@@ -849,7 +862,7 @@ export class NbSelectComponent
       return this.selectionModel.map((option: NbOptionComponent) => option.content).join(', ');
     }
 
-    return this.selectionModel[0].content;
+    return this.selectionModel[0]?.content;
```
Comment (Tibing) [inline src/framework/theme/components/select/select.component.ts:865]: why do we need this change?

File: src/framework/theme/components/select/select.component.ts [hunk 3]
```diff
@@ -927,6 +950,10 @@ export class NbSelectComponent
     if (this.isOpen) {
       this.ref.detach();
       this.cd.markForCheck();
+      this.selectClose.emit();
+
+      this.optionSearchInput.nativeElement.value = this.selectionView;
```
Comment (Tibing) [inline src/framework/theme/components/select/select.component.ts:955]: why don't we use FormControl for optionSearchInput?

**Review by Tibing — CHANGES_REQUESTED** [review]

**Comment by Tibing** [comment]
Also, let's add tests

File: src/framework/theme/components/select/select.component.ts (hunk 2 above)
Comment (sashaqred) [reply src/framework/theme/components/select/select.component.ts:865]: Earlier `selectionView` variable was used in the button only when something was selected. Now it is used in `optionSearchInput` all the time

File: src/framework/theme/components/select/select.component.html (hunk 1 above)
Comment (sashaqred) [reply src/framework/theme/components/select/select.component.html:2]: Just simpler. Nebular subscribes to some button events like focus. And with `*ngIf` we should implement unsubscription and resubscription mechanisms.  Looks too complicated for now

File: src/framework/theme/components/select/select.component.ts (hunk 3 above)
Comment (sashaqred) [reply src/framework/theme/components/select/select.component.ts:955]: `FormControl` adds some complexity to updating search input value when something in select is updated. Probably should be improved, but for now, I think it's not bad.

**Review by Tibing — APPROVED** [review]

**Review by Tibing — APPROVED** [review]

**Pull request merged**